### PR TITLE
#SPOI-6505 #resolve Added check of @OutputPortFieldAnnotation with no…

### PR DESCRIPTION
**Pull request by ishark**
… properties. One of the operators in Package had @OutputPortFieldAnnotation but did not have any properties in it.

### CODE REVIEW

File: engine/src/main/java/com/datatorrent/stram/webapp/asm/CompactUtil.java [hunk 1]
```diff
@@ -153,12 +153,14 @@ private static void setAnnotationNode(FieldNode fn, CompactFieldNode cfn) {
         if (annotation.desc.contains("InputPortFieldAnnotation")
             || annotation.desc.contains("OutputPortFieldAnnotation")) {
           List<Object> annotationValues = annotation.values;
-          int index = 0;
-          while (index < annotationValues.size()) {
-            annotationMap.put((String) annotationValues.get(index++), annotationValues.get(index++));
+          if (annotationValues != null) {
+            int index = 0;
+            while (index < annotationValues.size()) {
+              annotationMap.put((String) annotationValues.get(index++), annotationValues.get(index++));
```
Comment (ilganeli) [inline engine/src/main/java/com/datatorrent/stram/webapp/asm/CompactUtil.java]: I would suggest extracting the two `index++` to `String name = index++;` and `String val = index++;`

**Comment by tweise** [comment]
Is this needed for v3.2?

**Comment by ishark** [comment]
Yes, one of the operators in Malhar has port annotation but no annotation properties. And that is part of demo packages. So, uploading of package fails without this change.

**Comment by tweise** [comment]
OK, will cherry-pick it.

**Comment by ishark** [comment]
Thanks. Just squashed the commits.

**Comment by tweise** [comment]
In future, please use descriptive commit message with JIRA reference.

**Pull request merged**